src/pages/index.js

```diff
@@ -0,0 +1,327 @@
+const avatarLinkInput = avatarModal.querySelector("#profile-avatar-input");
+
+const deleteModal = document.querySelector("#delete-modal");
+
+const cardModal = document.querySelector("#add-card-modal");
+const cardForm = cardModal.querySelector(".modal__form");
+const cardSubmitBtn = cardModal.querySelector(".modal__submit-btn");
+const cardModalCloseBtn = cardModal.querySelector(".modal__close-btn");
+const cardLinkInput = cardModal.querySelector("#add-card-link-input");
+const cardNameInput = cardModal.querySelector("#add-card-name-input");
+
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+const previewModalCloseBtn = previewModal.querySelector(
+  ".modal__close-btn_type_preview"
+);
+
+const closeButtons = document.querySelectorAll(".modal__close-btn");
+const closeOverlayModals = document.querySelectorAll(".modal");
+
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  cardImageEl.addEventListener("click", () => {
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+    openModal(previewModal);
+  });
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  cardLikeBtn.addEventListener("click", () => {
+    const isLiked = cardLikeBtn.classList.contains("card__like-btn_liked");
+    if (isLiked) {
+      api
+        .changeLikeStatus(data._id)
+        .then((res) => {
+          cardLikeBtn.classList.remove("card__like-btn_liked");
+        })
+        .catch(console.error);
+    } else {
+      api
+        .changeLikeStatus(data._id)
+        .then((res) => {
+          cardLikeBtn.classList.add("card__like-btn_liked");
+        })
+        .catch(console.error);
+    }
+  });
+
+  cardDeleteBtn.addEventListener("click", () => {
+    openModal(deleteModal);
+    deleteModal.dataset.cardId = data._id;
+
+    cardElement.dataset.cardId = data._id;
+  });
+
+  return cardElement;
+}
+
+function handleDeleteConfirmation() {
+  const cardId = deleteModal.dataset.cardId;
+  const cardElement = document.querySelector(`.card[data-card-id="${cardId}"]`);
+
+  const deleteBtn = document.querySelector(".modal__submit-btn_type_delete");
```

Comment:
`  const deleteBtn = document.querySelector(".modal__submit-btn_type_delete");
`

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine